lib/sensitive_data_filter/middleware/parameter_parser.rb

```diff
@@ -27,10 +27,14 @@ def can_parse?(content_type)
 
       def parse(params)
         @parse.call params
+      rescue
```

Comment:
I thought about this, unfortunately, because parser can be defined by the users, we cannot know what parsing error is expected. We could update the config and instruct to raise a particular error. The user implementing the parsing would then have to catch exceptions in his parser and then re-raise them them as this particular exception. This feels cumbersome and relies on the user doing the right thing. Unfortunately an uncaught exception during parsing phase can create a major failure in an app, which may be able to handle parsing down the track.
If the parser defined by the user ONLY does what it is supposed to, i.e. parsing, then it is reasonable to assume that exceptions from that call would only relate to parsing. If parsing fails, the parameters will still be scanned, as unparsed strings, for sensitive data and cleansed.
What do you think we should do?